src/lightning/fabric/strategies/fsdp.py

```diff
@@ -185,6 +183,14 @@ def __init__(
         self.cpu_offload = _init_cpu_offload(cpu_offload)
         self.mixed_precision = mixed_precision
 
+    @property
+    def checkpoint_io(self) -> CheckpointIO:
+        raise NotImplementedError(f"The `{type(self).__name__}` does not use the `CheckpointIO` plugin interface.")
+
+    @checkpoint_io.setter
+    def checkpoint_io(self, io: Optional[CheckpointIO]) -> None:
```

Comment:
Why is None valid here? Shouldnt the getter also be annotated with optional then? And if the user sets None, should it also raise an error?

Reply:
Probably a mistake from the past. I updated it in all strategies.